cql/src/main/java/org/apache/cassandra/stargate/transport/internal/Server.java

```diff
@@ -590,12 +593,20 @@ private void send(InetAddressAndPort endpoint, Event.NodeEvent event) {
       // then don't send the notification. This covers the case of rpc_address set to "localhost",
       // which is not useful to any driver and in fact may cauase serious problems to some drivers,
       // see CASSANDRA-10052
-      if (!endpoint.equals(FBUtilities.getBroadcastAddressAndPort())
+      if (!areEqual(endpoint, FBUtilities.getBroadcastAddressAndPort())
```

Comment:
That looks like a legitimate bug, the two objects can't be equal because the types are not the same.